test/Microsoft.Framework.Caching.Memory.Tests/MemoryCacheSetAndRemoveTests.cs

```diff
@@ -209,5 +209,29 @@ public void RemoveAndReAddFromCallbackWorks()
             result = cache.Get(key);
             Assert.Same(obj2, result);
         }
+
+        [Fact]
+        public void SetGetAndRemoveWorks_WithNonStringKeys()
+        {
+            var cache = CreateCache();
+            var obj = new object();
+            var key = new Person { Id = 10, Name = "Mike" };
+
+            var result = cache.Set(key, obj);
+            Assert.Same(obj, result);
+
+            result = cache.Get(key);
+            Assert.Same(obj, result);
```

Comment:
@divega It works because it internally uses a Dictionary<> with the default equality comparer. Sent a PR with a regression test so that we don't break it.